eta/core/learning.py

```diff
@@ -82,6 +84,18 @@ def write_labels_map(labels_map, outpath):
             f.write("%s:%s\n" % (idx, labels_map[idx]))
 
 
+def get_class_labels(labels_map):
+    '''Returns the list of class labels from the given labels map.
+
+    Args:
+        a dictionary mapping indexes to label strings
+
+    Returns:
+        a list of class labels
+    '''
+    return [labels_map[k] for k in sorted(labels_map.keys())]
```

Comment:
What about circumstances where the labels map is 1-based instead of 0-based?

It seems like this function makes an assumption and as such should have a check that the keys are incrementing integers starting at 0

Reply:
I updated this to ensure that all IDs between `min(1,min(labels_map))` and `max(labels_map)` will have labels generated. This handles cases where there are missing labels within the map, which, annoyingly, *IS* true for some COCO label maps (at one point the list was expanded from 80 to 91 labels).

0 vs 1-based indexing doesn't matter here- either will work. Its the user's responsibility to ensure that this list aligns with any vectors of class probabilities they generate

Reply:
yes, technically, if one is using 0-based indexing then `min(1,min(labels_map))` should be `min(0,min(labels_map))`, but its silly to provide a label map that doesn't contain the min or max label, so I'm not worried about that edge case